src/lib.rs

```diff
@@ -51,9 +57,295 @@ mod writer;
+        }
+    }
+}
+
+impl FromStr for KeyMethod {
+    type Err = ParseEnumError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "clear" => Ok(KeyMethod::Clear),
+            "base64" => Ok(KeyMethod::Base64),
+            "uri" => Ok(KeyMethod::Uri),
+            "prompt" => Ok(KeyMethod::Prompt),
+            _ => Err(ParseEnumError::Invalid(s.to_string())),
+        }
+    }
+}
+
+impl Display for KeyMethod {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(self.as_str())
+    }
+}
+
+/// The media type
+///
+/// See [RFC 8866 Section 5.14](https://datatracker.ietf.org/doc/html/rfc8866#section-5.14)
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum MediaType {
+    // Audio type
+    Audio,
+    // Video type
+    Video,
+    // Text type
+    Text,
+    // Application type
+    Application,
+    // Message type
+    Message,
+    // Image type. See [RFC 6466](https://datatracker.ietf.org/doc/html/rfc6466)
+    Image,
+}
+
+impl MediaType {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            MediaType::Audio => "audio",
+            MediaType::Video => "video",
+            MediaType::Text => "text",
+            MediaType::Application => "application",
+            MediaType::Message => "message",
+            MediaType::Image => "image",
+        }
+    }
+}
+
+impl FromStr for MediaType {
+    type Err = ParseEnumError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "audio" => Ok(MediaType::Audio),
+            "video" => Ok(MediaType::Video),
+            "text" => Ok(MediaType::Text),
+            "application" => Ok(MediaType::Application),
+            "message" => Ok(MediaType::Message),
+            "image" => Ok(MediaType::Image),
+            _ => Err(ParseEnumError::Invalid(s.to_string())),
+        }
+    }
+}
+
+impl Display for MediaType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(self.as_str())
+    }
+}
+
+/// Transport Protocol for the media
+///
+/// See [RFC 8866 Section 5.14](https://datatracker.ietf.org/doc/html/rfc8866#section-5.14)
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum TransportProto {
+    // Direct UDP
+    Udp,
+    // RTP over UDP
+    RtpAvp,
+    // Secure RTP over UDP
+    RtpSavp,
+    // Secure RTP over UDP with RTCP-based feedback
+    RtpSavpf,
+}
+
+impl TransportProto {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            TransportProto::Udp => "udp",
+            TransportProto::RtpAvp => "RTP/AVP",
+            TransportProto::RtpSavp => "RTP/SAVP",
+            TransportProto::RtpSavpf => "RTP/SAVPF",
+        }
+    }
+}
+
+impl FromStr for TransportProto {
+    type Err = ParseEnumError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "audio" => Ok(TransportProto::Udp),
```

Comment:
```suggestion
            "udp" => Ok(TransportProto::Udp),
```

?

Are these supposed to be case-insensitive maybe?

Reply:
Yes, changed to ignore the case